osu.Game/Rulesets/Mods/ModRateChanger.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Audio;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Localisation;
+using osu.Framework.Logging;
+using osu.Game.Configuration;
+using osu.Game.Graphics;
+
+namespace osu.Game.Rulesets.Mods
+{
+    public abstract class ModRateChanger : ModRateAdjust
+    {
+        public override string Name => "Rate Changer";
+        public override string Acronym => "RC";
+        public override IconUsage? Icon => OsuIcon.ModHalftime;
+        public override ModType Type => ModType.Conversion;
+        public override LocalisableString Description => "Change speed and pitch separately for any song";
+
+        private readonly BindableNumber<double> tempoAdjust = new BindableDouble(1);
+
+        private readonly BindableNumber<double> freqAdjust = new BindableDouble(1);
+
+        public double speedMultiplier = 1;
+
+        bool allowLog = true;
+        
+        [SettingSource("Speed edit", "The song speed multiplier")]
+        public override BindableNumber<double> SpeedChange { get; } = new BindableDouble(1)
+        {
+            MinValue = 0.1,
+            MaxValue = 2,
+            Precision = 0.01,
+        };
+
+        [SettingSource("Pitch edit", "The song pitch multiplier")]
+        public BindableNumber<double> PitchChange { get; } = new BindableDouble(1)
+        {
+            MinValue = 0.1,
+            MaxValue = 2,
+            Precision = 0.01,
+        };
+
+        protected ModRateChanger()
+        {
+            SpeedChange.BindValueChanged(val =>
+            {
+                speedMultiplier = val.NewValue;
+
+                ValTempoReal();
+            }, true);
+
+            PitchChange.BindValueChanged(val =>
+            {
+                freqAdjust.Value = val.NewValue;
+
+                ValTempoReal();
+            }, true);
+        }
+
+        public override void ApplyToTrack(IAdjustableAudioComponent track)
+        {
+            // base.ApplyToTrack() intentionally not called (different tempo adjustment is applied)
+            track.AddAdjustment(AdjustableProperty.Frequency, freqAdjust);
+            track.AddAdjustment(AdjustableProperty.Tempo, tempoAdjust);
+        }
+        
+        // Implements 1/x to define the value of tempoAdjust.
+        void ValTempoReal() {
+            if ((speedMultiplier * (1/freqAdjust.Value)) > 0.05)
```

Comment:
this cannot be a thing:

https://user-images.githubusercontent.com/74463310/221634994-0a08af7b-64a9-47e1-830f-b3f8381269a1.mov

Reply:
Removed that log part entirely now but, could you tell me what would've been the optimal way to make it only show a single message in this context?

Reply:
Well the thing is mods should simply not be allowing impossible combinations. So if you had to limit it somehow I'd say like make the sliders nub ( the part you drag) flash red or something of the sort and refuse to move further. You can see in the wind up mods the nubs push eachother to avoid impossible combinations. Something similar here could be done .

Reply:
Nice, that's a very good suggestion.